gui/gui/src/TRootGuiFactory.cxx

```diff
@@ -49,17 +51,48 @@ TApplicationImp *TRootGuiFactory::CreateApplicationImp(const char *classname,
    TRootApplication *app = new TRootApplication(classname, argc, argv);
    if (!app->Client()) {
       delete app;
-      app = 0;
+      app = nullptr;
    }
    return app;
 }
 
+////////////////////////////////////////////////////////////////////////////////////////
+/// Show warning that TWebCanvas will be started by default
+
+void TRootGuiFactory::ShowWebCanvasWarning()
+{
+   static bool show_warn = true;
+   if (!show_warn || gROOT->IsWebDisplay()) return;
+   show_warn = false;
+
+   std::cout << "\n"
+                "   !!! ATTENTION !!! \n"
+                "\n"
+                "ROOT comes with a web-based canvas, which is now being started. \n"
+                "Revert to default by setting \"Canvas.Name: TRootCanvas\" in rootrc file or\n"
+                "by starting \"root --web=off\". Web-based TCanvas can be used in batch mode for\n"
+                "image generation when running with \"root -b --web\"\n"
```

Comment:
```suggestion
                "by starting \"root --web=off\".\n"
```
I don't think this is the place for implementation details / documentation. Instead, this information should be shown in the blog post you might author on the new default TCanvas.